src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,25 @@
+package duke;
+
+import duke.command.*;
+
+public class Parser {
+    public static Command parse(String input) throws DukeException {
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("done")) {
+            return new DoneCommand(input);
+        } else if (input.startsWith("delete")) {
+            return new DeleteCommand(input);
+        } else if (input.startsWith("todo")) {
+            return new TodoCommand(input);
+        } else if (input.startsWith("deadline")) {
+            return new DeadlineCommand(input);
+        } else if (input.startsWith("event")) {
+            return new EventCommand(input);
```

Comment:
Using `input.startsWith("...")` may cause some invalid commands to be interpreted as valid commands, e.g. when input starts with `deadliner` or `todoy`. Instead, consider using `input.split(" ", 2)` to get a `String[]`, then check equality against the first element of the `String[]`. This way, the second element of the `String[]` also contains all the information you need for your `Command` subclasses later on.